handout/zones/README.md

```diff
@@ -0,0 +1,132 @@
+# Zones
+
+[Zone.js](https://github.com/angular/zone.js) provides a mechanism, called zones, for encapsulating and intercepting asynchronous activities in the browser (e.g. setTimeout, XHR requests, promises).
+
+These zones are *execution contexts* that allow Angular to track the start and completion of asynchronous activities and perform tasks as required (e.g. change detection). Zone.js provides a global zone that can be forked and extended to further encapsulate/isolate asynchronous behaviour, which Angular does so in its **NgZone** service, by creating a fork and extending it with its own behaviours.
+
+The **NgZone** service provides us with a number of Observables and methods for determining the state of Angular's zone and to execute code in different ways inside and outside Angular's zone.
+
+It is important to know that Zone.js accomplishes these various interceptions by [Monkey Patching](https://en.wikipedia.org/wiki/Monkey_patch) common methods and elements in the browser, e.g. setTimeout and HTMLElement.prototype.onclick. These interceptions can cause unexpected behaviour between external libraries and Angular. In some cases, it may be preferential to execute third party methods outside of Angular's zone (see below).
```

Comment:
Using backticks for `setTimeout` and `HTMLElement.prototype.onclick` might improve readability